pavelib/quality.py

```diff
@@ -147,10 +142,21 @@ def run_pylint(options):
     with open(Env.METRICS_DIR / "pylint", "w") as f:
         f.write(violations_count_str)
 
-    # Fail number of violations is greater than the limit
-    if num_violations > violations_limit > -1:
-        raise BuildFailure("Failed. Too many pylint violations. "
-                           "The limit is {violations_limit}.".format(violations_limit=violations_limit))
+    # Fail when number of violations is less than the lower limit,
+    # which likely means that pylint did not run successfully.
+    if num_violations < lower_violations_limit > -1:
```

Comment:
Couple of thoughts:
1. Since this is just to trap for problems, why not just have some calculation based off the limit?  Could be 75% just to throw out anything as self-adjusting protection.  The problems with making it a parameter are: a) someone needs to think about it when they are lowering the threshold and, b) someone needs to understand its purpose, which is odd to document and easy to wonder: "why are too few violations a problem"?
2. Consider adding two cases/messages here:
```
if (num_violations == 0 and upper_violations_limit > 0):
  # Ask if pylint was mistakenly disabled

if num_violations < lower_violations_limit > -1:
  # State that the pylint violations were low, which either indicates that the threshold should be lowered or pylint is not running completely.

Reply:
Thanks for the review!
1. I added more comments around the intent of this lower limit. I prefer a direct approach of setting a lower limit (if desired) than guessing as a fraction. This limit will not be something that changes often.
2. The lower limit takes care of the first case.  The second case already exists and I added more comments and printed text describing the situation.